runtime/service.go

```diff
@@ -769,6 +785,142 @@ func (s *service) GetVMMetadata(requestCtx context.Context, request *proto.GetVM
 	return &proto.GetVMMetadataResponse{Metadata: string(metadata)}, nil
 }
 
+func (s *service) createBalloon(requestCtx context.Context, request *proto.CreateVMRequest) (models.Balloon, error) {
+	amountMB := request.BalloonDevice.AmountMb
+	deflateOnOom := request.BalloonDevice.DeflateOnOom
+	statsPollingIntervals := request.BalloonDevice.StatsPollingIntervals
+
+	balloonDevice := firecracker.NewBalloonDevice(amountMB, deflateOnOom, firecracker.WithStatsPollingIntervals(statsPollingIntervals))
+	balloon := balloonDevice.Build()
+
+	if balloon.AmountMb == nil || balloon.DeflateOnOom == nil {
+		return balloon, errors.Errorf("One of balloon properties is nil, please check AmountMb, DeflateOnOom")
+	}
```

Comment:
If these checks are coming from Firecracker, we probably should have them on the Go SDK.

Reply:
Fixed it.

Reply:
Fixed it.